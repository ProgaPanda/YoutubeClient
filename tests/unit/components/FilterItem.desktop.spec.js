import { shallowMount } from '@vue/test-utils';
import FilterItemDesktop from '@/components/FilterItem.desktop.component.vue';

describe('FilterItemDesktop', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(FilterItemDesktop, {
      propsData: {
        name: 'Type',
        options: ['All', 'Video', 'Channel', 'Playlist'],
        defaultOption: 'All',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correct number of filter options', () => {
    const wrapper = shallowMount(FilterItemDesktop, {
      propsData: {
        name: 'Type',
        options: ['All', 'Video', 'Channel', 'Playlist'],
        defaultOption: 'All',
      },
    });

    expect(wrapper.find('.filter-item__list').findAll('.filter-item__list__item')).toHaveLength(4);
  });
});
