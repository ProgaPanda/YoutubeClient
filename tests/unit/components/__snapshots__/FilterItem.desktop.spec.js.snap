// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItemDesktop renders correctly 1`] = `
<div
  class="filter-item"
>
  <h5
    class="filter-item__title"
  >
    Type
  </h5>
   
  <ul
    class="filter-item__list"
  >
    <li
      class="filter-item__list__item filter-item__list__item--active"
    >
      
      All
      
      <!---->
    </li>
    <li
      class="filter-item__list__item"
    >
      
      Video
      
      <!---->
    </li>
    <li
      class="filter-item__list__item"
    >
      
      Channel
      
      <!---->
    </li>
    <li
      class="filter-item__list__item"
    >
      
      Playlist
      
      <!---->
    </li>
  </ul>
</div>
`;
